ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.view.js

```diff
@@ -235,23 +251,19 @@ module.exports = Marionette.LayoutView.extend({
     var currentComparator = this.model.get('comparator')
     switch (propertyJSON.type) {
       case 'LOCATION':
-        if (['INTERSECTS'].indexOf(currentComparator) === -1) {
+        if (geometryComparators.indexOf(currentComparator) === -1) {
           this.model.set('comparator', 'INTERSECTS')
         }
-        this.toggleLocationClass(true)
+        this.toggleLocationClass(currentComparator === 'IS EMPTY')
```

Comment:
Should this be true for either case? This won't toggle the location class if the comparator is `INTERSECTS`

Reply:
This is to get rid of the text area box when choosing a location related attribute, but you only want to do this when the `IS EMPTY` operator is chosen.

Reply:
Ah and it was always true before because we made assumptions about the state it was in when we called toggle on it. I think this should fall into the other fixes to stop using toggle on elements.